### PR TITLE
[HIPIFY][doc] Building Documentation on RTD

### PR DESCRIPTION
**Pull request by saadrahim**
Summary of proposed changes:
- Updates the readthedocs documentation to use the rocm-docs-core template

Rationale:
 - https://github.com/RadeonOpenCompute/rocm-docs-core#purpose
 - https://confluence.amd.com/display/MLSE/Docs+as+Code+Transition
 - https://confluence.amd.com/display/MLSE/Documentation+RACI+Page

Example:
https://advanced-micro-devices-demo--737.com.readthedocs.build/projects/HIPIFY/en/737/hipify-clang.html

### CODE REVIEW

File: README.md
Comment (emankov) [inline README.md:19]: Why the tocstop is removed?

File: README.md
Comment (emankov) [inline README.md]: Why only rocBLAS is mentioned? What about all other supported libs?

File: docs/index.md
Comment (emankov) [inline docs/index.md]: Please provide the text here.

File: README.md
Comment (emankov) [inline README.md]: The page is missing.

File: README.md
Comment (emankov) [inline README.md]: Unneeded line break.

File: README.md
Comment (emankov) [inline README.md]: This subparagraph should be under `Linux` title.

File: README.md
Comment (emankov) [inline README.md]: Unneeded line break.

File: README.md
Comment (emankov) [inline README.md]: [4] Outdated and unrelated to HIPIFY tools.

File: README.md
Comment (emankov) [inline README.md]: Excessive space in `[2]  C++`.

**Review by emankov — CHANGES_REQUESTED** [review]

**Comment by saadrahim** [comment]
Apologies, please don't review it yet. I created the PR just to get an automatic builds working. Please consider this as a draft for now.

File: README.md
Comment (saadrahim) [reply README.md:19]: The documentation has been reorganized such that the readme is not the starting point for documentation navigation. Users are directed to the website. The toc is handled there.

File: README.md
Comment (saadrahim) [reply README.md]: Please see new documentation page

**Comment by saadrahim** [comment]
This is almost ready to review except for some numbering issues that need fixing.

**Comment by saadrahim** [comment]
@emankov Please take a look at the rendered documentation in the "All checks have passed" section. There are still a few issues with table formatting, mainly table width. I am working on the table width issues.

File: README.md
Comment (emankov) [reply README.md:19]: The majority of the users are still looking at README.md which should have its own tocstop. Here is the biweekly stats:

image

Comment (saadrahim) [reply README.md:19]: I added a link from the README into the documentation source at index.md. I am hoping that will be a compromise on removing the TOC. 
The github repository will link to the documentation site by configuring the information below.
image
Overtime, the documentation portal will feature as high in the search engines than github repository. The intention is too direct users to our documentation portal to consume our documents. The documentation portal will contain links to the source repository and enable users to look at the doc source on github. Github will always remain the source of truth.

**Comment by emankov** [comment]
I don't quite understand how README.md will be presented on GitHub after the merge.

File: docs/source/tables/CUBLAS_API_supported_by_HIP.md
Comment (emankov) [inline docs/source/tables/CUBLAS_API_supported_by_HIP.md]: The entire CUDA2HIP documentation is autogenerated by hipify-clang. We already agreed on paragraph numbering with Roopa, and this is how these docs are presented on docs.amd.com.

**Review by emankov — CHANGES_REQUESTED** [review]
Please, revert changes in the `*_supported_by_HIP.md` files - they are autogenerated by `hipify-clang` and shouldn't be changed manually.

**Comment by saadrahim** [comment]
> Please, revert changes in the `*_supported_by_HIP.md` files - they are autogenerated by `hipify-clang` and shouldn't be changed manually.

@emankov I will need to request changes to the autogeneration of the *_supported_by_HIP.md files. Please expect the requirements in a ticket soon. The usage of header bold formatting needs to be removed. Also, numbering of documentation headers needs to be handled by sphinx, the publishing tool. Your repository was a test bed for formatting documentation. The requirements for formatting markdown files will be published as part of documentation formatting guidelines for ROCm within a few weeks. I hope to address your concerns with the formatting guidelines.

File: docs/source/conf.py
Comment (saadrahim) [inline docs/source/conf.py]: extra line break?

File: docs/source/requirements.in
Comment (saadrahim) [inline docs/source/requirements.in]: Let's move to ROCm_Documentation

Comment (saadrahim) [reply docs/source/requirements.in]: https://github.com/RadeonOpenCompute/ROCm_Documentation using the main branch.

**Comment by saadrahim** [comment]
> Please, revert changes in the `*_supported_by_HIP.md` files - they are autogenerated by `hipify-clang` and shouldn't be changed manually.

@emankov I've reverted these changes. 

I am still conflicted with the structure of the README. Is ok to have a condensed README for now? Let's discuss via Mark on what a README for a project should have versus documentation.

I would like to merge these changes in soon. This PR has been open for more than a month now. Hopefully, I've addressed your concerns.

**Comment by samjwu** [comment]
Updated/standardized HIPIFY docs to use latest rocm-docs-core templates

Successful Build: https://readthedocs.com/projects/advanced-micro-devices-hipify/builds/1358769/

Built Docs: https://advanced-micro-devices-demo--737.com.readthedocs.build/projects/HIPIFY/en/737/index.html

File: docs/index.md
Comment (emankov) [inline docs/index.md]: A typo: perl, not pearl

File: README.md
Comment (emankov) [reply README.md:19]: > Overtime, the documentation portal will feature as high in the search engines than github repository.

I don't think so; I don't believe it'll ever happen. But it doesn't matter, as `README.md` is left alive in its place with your changes.

> Github will always remain the source of truth.

I agree it should be so, but I see that with the change we'll have four md files instead of a single `README.md`:
- A bit changed `README.md` - *Do not see any need for changing (especially stripping) it at all if it is for GitHub repo only.
- `hipify-clang.md`
- `hipify-perl.md`
- `supported_apis.md`

The information in the last three files is taken from `README.md` (duplicates).

Could you please explain to us how those last three are produced and how they are going to be supported?

Comment (emankov) [reply README.md:19]: It looks like TOC in `README.md` can be left untouched without harm to the generated documentation. Correct me, if I'm wrong.

File: README.md
Comment (emankov) [inline README.md]: It looks to me strange, as HIPIFY documentation provided by the link duplicates the README.md itself. Maybe I missed smth?

File: README.md
Comment (emankov) [inline README.md]: This subparagraph should be moved to the Linux paragraph.

File: docs/.sphinx/_toc.yml.in
Comment (emankov) [inline docs/.sphinx/_toc.yml.in:23]: Should it be supported explicitly?

Comment (emankov) [reply docs/.sphinx/_toc.yml.in:23]: **[ToFix]**
The paths to CUDA2HIP docs are changed, so the corresponding paths in the links of `README.md`  (#-supported-cuda-apis) have to be changed as well.

**Review by emankov — CHANGES_REQUESTED** [review]

File: docs/index.md
Comment (samjwu) [reply docs/index.md]: Fixed in [7f7fe87](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/7f7fe87c7db9e19ecf791cf2c30e5ce8a3c8ea4d)

File: README.md
Comment (samjwu) [reply README.md]: Good point. I have replaced the line about the documentation source with instructions to build the docs locally.
[6f7acbb](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/6f7acbb5362f921fb477393e4b770dd017cc9e4d)

File: README.md
Comment (samjwu) [reply README.md:19]: Apologies, I restored the README.md when resolving merge conflicts, but this was not intentional. As the current README is relatively long, it seems easier to read on a site dedicated to documentation and link to the site in the README. The goal is to have the [Read the Docs site](https://advanced-micro-devices-demo--737.com.readthedocs.build/projects/HIPIFY/en/737/hipify-clang.html) replace the specific details in the README (see link for example).

If you agree, I propose the content below `<!-- tocstop -->` in the README be removed. This also prevents the problem you mentioned about having to support both the Read the Docs .md files and the README since it removes the duplicated content in the README.
This includes removing the hipify-clang/perl and supported CUDA API docs (migrated to Read the Docs), the disclaimer (generic/standard legal statements and licenses will be handled by rocm-docs-core in the future), and the TOC/table of contents (see the sidebar on the [RTD site](https://advanced-micro-devices-demo--737.com.readthedocs.build/projects/HIPIFY/en/737/supported_apis.html)).

Comment (emankov) [reply README.md:19]: Hello, @samjwu,

> I restored the README.md when resolving merge conflicts, but this was not intentional

Does it mean the `README.md` won't be available in the repo?

>As the current README is relatively long, it seems easier to read on a site dedicated to documentation and link to the site in the README.

I don't think the relative length of the `README.md` is a strong reason here. More details - fewer mistakes - lesser issues like `doesn't work for me` reported, you know.

**Let me suggest yet another approach instead.**

1. We'll leave `README.md` almost untouched with its own TOC, but the contents will be constituted from `hipify-clang.md`, `hipify-perl.md`, and `supported_apis.md` if you wish so.
2. The TOC links will be changed to the corresponding `#` headers in those three files. Smth like follows:
```
- [hipify-clang](./docs/hipify-clang.md#clang)
     * [Dependencies](./docs/hipify-clang.md#dependencies)
     * [Usage](./docs/hipify-clang.md#hipify-clang-usage)
     * [Building](./docs/hipify-clang.md#building)
     * [Testing](./docs/hipify-clang.md#testing)
     * [Linux](./docs/hipify-clang.md#linux)
     * [Windows](./docs/hipify-clang.md#windows)
- [hipify-perl](./docs/hipify-perl.md#perl)
     * [Usage](./docs/hipify-perl.md#hipify-perl-usage)
     * [Building](./docs/hipify-perl.md#hipify-perl-building)
- [Supported CUDA APIs](./docs/supported_apis.md)
- [Disclaimer](#disclaimer)
```
3. The three child .md files should contain exact parts of the original `README.md` without any semantical or syntactical changes; the headers should be preserved as well.

Comment (samjwu) [reply README.md:19]: Works for me. I'll add back the TOC in the README and make those 3 .md files mentioned identical to the README for the relevant sections.

File: README.md
Comment (samjwu) [reply README.md]: Fixed in latest [47e9926](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/47e9926058e7fe40d09fe4b083dff7b95296bcce)

File: README.md
Comment (samjwu) [reply README.md]: Fixed in latest [47e9926](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/47e9926058e7fe40d09fe4b083dff7b95296bcce)

File: README.md
Comment (samjwu) [reply README.md]: Fixed in latest [47e9926](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/47e9926058e7fe40d09fe4b083dff7b95296bcce)

File: README.md
Comment (samjwu) [reply README.md]: Fixed in latest [47e9926](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/47e9926058e7fe40d09fe4b083dff7b95296bcce)

File: docs/.sphinx/_toc.yml.in
Comment (samjwu) [reply docs/.sphinx/_toc.yml.in:23]: Fixed in [c684bf8](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/c684bf8b46240827e5246399513416694802ca45)

File: README.md
Comment (samjwu) [reply README.md:19]: [47e9926](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/47e9926058e7fe40d09fe4b083dff7b95296bcce) - implements points 1, 2, and 3

[c684bf8](https://github.com/ROCm-Developer-Tools/HIPIFY/pull/737/commits/c684bf8b46240827e5246399513416694802ca45) - updates links to docs in README and docs pages. also makes a couple changes to headers in docs pages for the RTD website links to work correctly

Comment (saadrahim) [reply README.md:19]: Hi @emankov,

There needs to be some changes to the content on your README.md. @samjwu reverted many of my changes. They need to be restored. The new files created in docs/ are intended to move the contents of your README to a new location. The README in this project needs adopt README standards according to GitHub's README guidelines at https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes. Additional requirements to the README structure may be mandated by ROCm's documentation team to standardize the look and feel. The README is not a single file documentation for the entire project. It is intended to point people to the documentation and other resources, if any, that can help the user understand how to use HIPIFY. HIPIFY documentation will be hosted online on a webserver. The webserver based documentation should be the first recommended option for users to consume documentation. The README may also link to the markdown files in the repository.

Comment (emankov) [reply README.md:19]: Hello @saadrahim,

I didn't ask to revert; I just suggested yet another approach, and I do not see contradictions between it and your post. Can we discuss it offline? We have a corresponding thread.

Regards,
Evgeny

**Comment by saadrahim** [comment]
@emankov hoping you can approve this PR. @samjwu or I will resolve the merge conflicts.

**Review by emankov — APPROVED** [review]
LGTM, let's merge and see what it'll look like.

**Comment by emankov** [comment]
@samjwu, I wanted to finish with fixing your #737 and merge both simultaneously, because README.md is broken. Please have a look. And also, please, do not merge your doc-related changes in the future. Thanks.

**Pull request merged**
